### PR TITLE
feat: increase coverage of geometry domain for use with textbook problems

### PR DESCRIPTION
**Pull request by hsharriman**
# Description
Updated the geometry domain, euclidean style, and relevant Penrose files to be able to visualize diagrams from a high school geometry textbook. Implemented 24 working examples of various Euclidean geometry diagrams in the `examples/geometry-domain` folder. Also, replaced outdated versions of `euclidean.sty` and `geometry.dsl` as well as broken `geometry` examples from `examples/` folder and replaced them with the up-to-date versions that were used to create the textbook problems.

# Implementation strategy and design decisions
A few of the predicates behaviors are blocked by #629. Once this is merged I can open a new PR to implement things like multiple tick marks for a segment.

Generally speaking, I tried to keep as much of the calculation logic inside of style as possible, only implementing new Functions and constraints if it was absolutely necessary.

# Examples with steps to reproduce them
* `yarn start`
* To run a given example (`c11p12 in this case, which demonstrates circles, radii, chords, and angle bisectors), run: `npx roger watch examples/geometry-domain/textbook_problems/c11p12.sub examples/geometry-domain/euclidean.sty  examples/geometry-domain/geometry.dsl`
* visit `localhost:3000` Depending on the example you might need to resample a few times.
# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass locally using `yarn test`
- [x] I ran `yarn docs` and there were no errors when generating the HTML site
- [x] My code follows the style guidelines of this project (e.g.: no ESLint warnings)

# Open questions

Questions that require more discussion or to be addressed in future development:

### CODE REVIEW

**Comment by wodeni** [comment]
I used `automator` to run the registry. Looking great! Here's how I did it:

* Go to `packages/automator`
* Run `yarn start batch registry.json out/ --src-prefix=../../examples/geometry-domain  --folders` to generate all Penrose diagrams
* Run `yarn start render out browser` to generate the static site.

Adding some summary stats from rendering the registry in `textbook-problems`:
visualization

File: packages/core/src/contrib/Functions.ts [hunk 1]
```diff
@@ -429,6 +425,21 @@ export const compDict = {
       contents: ifCond(gt(cross, varOf(0)), varOf(0), varOf(1)),
     };
   },
+  /**
+   * Return a point located at the midpoint of a line `s1`
+   */
+  midpoint: ([t1, s1]: [string, any]): IVectorV<VarAD> => {
```
Comment (wodeni) [inline packages/core/src/contrib/Functions.ts:431]: To make it fully general, can we make this function take in two points instead?

**Review by wodeni — COMMENTED** [review]
Just finished a light pass on the TypeScript code. Looks good to me overall. I'll take a closer look at the Style program in the next round.

File: packages/core/src/contrib/Constraints.ts [hunk 2]
```diff
@@ -595,6 +598,19 @@ export const constrDict = {
...
+  ptCircleIntersect: (p: VarAD[], [t, s]: [string, any]) => {
+    if (t === "Circle") {
+      const r = s.r.contents;
+      const c = fns.center(s);
+      return squared(sub(ops.vdist(p, c), r));
+    } else throw new Error(`${t} not supported for ptCircleIntersect`);
+  },
+
+  ptLineIntersect: (p: VarAD[], [t, s]: [string, any]) => {
+    if (isLinelike(t)) {
```
[the remark this answers is not in the document]
Comment (hsharriman) [reply packages/core/src/contrib/Constraints.ts]: Turns out I don't need this function for anything so I just removed it for the time being.

**Review by wodeni — APPROVED** [review]

**Review by wodeni — COMMENTED** [review]
LGTM! There seem to be some conflicts with `main`. Should be good to go once resolved :P

**Pull request merged**